# Week 03 — Function 05

## Table of contents

1. [Overview](#overview)
2. [Objectives](#objectives)
3. [Evidence provenance](#evidence-provenance)
4. [Environment and setup](#environment-setup)
5. [Data validation](#data-validation)
6. [Descriptive EDA](#descriptive-eda)
7. [Visual EDA](#visual-eda)
8. [Model and acquisition](#model-acquisition)
9. [Week 3 proposal](#week-3-proposal)
10. [Reproducibility checks](#reproducibility-checks)
11. [Conclusions and next steps](#conclusions-next-steps)

<a id="overview"></a>
## 1. Overview

This focused review mirrors the canonical Week 3 methodology for Function 05, with Weeks 1–2 observed and Week 3 proposed only.

<a id="objectives"></a>
## 2. Objectives

Validate the 4-dimensional evidence, assess the latest returned point, and reproduce the recorded Manual reasoning proposal without look-ahead.

<a id="evidence-provenance"></a>
## 3. Evidence provenance

Starter arrays come from `Week_01/Function_nn/03_Data`; exact returned pairs come from `Results/query_output_ledger.csv`. The Week 3 return is excluded because it was unknown when the proposal was selected.

<a id="environment-setup"></a>
## 4. Environment and setup

In [1]:
from pathlib import Path
import sys
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

ROOT=Path.cwd().resolve()
for candidate in (ROOT,*ROOT.parents):
    if (candidate/'Week_03'/'Function_05').is_dir(): ROOT=candidate; break
else: raise FileNotFoundError('Could not locate repository root')
if str(ROOT) not in sys.path: sys.path.insert(0,str(ROOT))
from Code.historical_function_review import analyse_historical_function

<a id="data-validation"></a>
## 5. Data validation

In [2]:
observations, summary, proposal, diagnostic_figure = analyse_historical_function(3, 5, ROOT)
input_columns=[f'x{i}' for i in range(1,5)]
inputs=observations[input_columns].to_numpy(float)
outputs=observations['objective'].to_numpy(float)
assert inputs.shape==(22,4) and outputs.shape==(22,)
assert np.isfinite(inputs).all() and np.isfinite(outputs).all()
assert np.all((inputs>=0)&(inputs<=1))
observations

,query,evidence,x1,x2,x3,x4,objective
0,1,starter,0.191447,0.038193,0.607418,0.414584,64.443440
1,2,starter,0.758653,0.536518,0.656000,0.360342,18.301380
2,3,starter,0.438350,0.804340,0.210245,0.151295,0.112940
3,4,starter,0.706051,0.534192,0.264243,0.482088,4.210898
4,5,starter,0.836478,0.193610,0.663893,0.785649,258.370525
5,6,starter,0.683432,0.118663,0.829046,0.567577,78.434389
6,7,starter,0.553621,0.667350,0.323806,0.814870,57.571537
7,8,starter,0.352356,0.322242,0.116979,0.473113,109.571876
8,9,starter,0.153786,0.729382,0.422598,0.443074,8.847992
9,10,starter,0.463442,0.630025,0.107906,0.957644,233.223610


<a id="descriptive-eda"></a>
## 6. Descriptive EDA

All comparisons are descriptive and within-function; no causal, global-optimum, or cross-function ranking claim is made.

In [3]:
pd.Series({k:v for k,v in summary.items() if k!='proposal'}, name='verified evidence')

week                                                                             3
function                                                                         5
dimensions                                                                       4
starter_observations                                                            20
recorded_pairs                                                                   2
total_verified_observations                                                     22
best_query                                                                      16
best_input                       [0.22418902330288348, 0.8464804904862864, 0.87...
best_output                                                            1088.859618
latest_verified_query                                                           22
latest_verified_input                      [0.255841, 0.841692, 0.888984, 0.86026]
latest_verified_output                                                 1035.634146
late

<a id="visual-eda"></a>
## 7. Visual EDA

Orange markers are returned Weeks 1–2 observations; the star is the verified incumbent. The Week 3 proposal is deliberately absent.

In [4]:
display(diagnostic_figure)
plt.close(diagnostic_figure)

<Figure size 1200x450 with 3 Axes>

<a id="model-acquisition"></a>
## 8. Model and acquisition

Method: **Manual reasoning**. This adaptive policy is a heuristic chosen from the evidence available at the decision boundary; it is not a statistically controlled acquisition comparison.

<a id="week-3-proposal"></a>
## 9. Week 3 proposal

Proposed only: `[0.255842, 0.841692, 0.888985, 0.860266]`. Decision record: Chosen using evidence through Week 2, before the Week 3 return.

<a id="reproducibility-checks"></a>
## 10. Reproducibility checks

In [5]:
candidate=np.asarray(proposal['query'],dtype=float)
assert proposal['status']=='proposed_only'
assert candidate.shape==(4,) and np.all((candidate>=0)&(candidate<=0.999999))
duplicate=bool(np.any(np.all(np.isclose(inputs,candidate,rtol=0,atol=5e-7),axis=1)))
assert duplicate==proposal['duplicates_observed_evidence']
assert summary['recorded_pairs']==2
portal='-'.join(f'{value:.6f}' for value in candidate)
assert all(len(part.split('.')[-1])==6 for part in portal.split('-'))
print('Function 05 Week 3 checks passed:', portal, 'duplicate:', duplicate)

Function 05 Week 3 checks passed: 0.255842-0.841692-0.888985-0.860266 duplicate: False


<a id="conclusions-next-steps"></a>
## 11. Conclusions and next steps

The evidence boundary is locked at 22 verified observations. The Week 3 proposal remains unobserved until its authoritative return is appended at the next checkpoint.